### PR TITLE
MBM-392 Refactor Registry

### PR DESCRIPTION
**Pull request by insertStringHere**

### CODE REVIEW

**Review by MattKirchh — APPROVED** [review]

File: src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java [hunk 1]
```diff
@@ -236,4 +239,15 @@ protected static NonNullConsumer<PlayerMana> handlePlayerMana(Player player) {
...
+    @SubscribeEvent
+    public static void onPlayerCloned(PlayerEvent.Clone event){
+        if(event.isWasDeath() && ManaConfig.keepArmor.get()){
+            for(ItemStack item : event.getOriginal().getArmorSlots()){
+                if(!item.isEmpty() && item.getItem() instanceof MyrialArmorItem){
+                    event.getPlayer().setItemSlot(Player.getEquipmentSlotForItem(item), item);
+                }
+            }
+        }
+    }
```
Comment (MattKirchh) [inline src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java:252]: Isn't this also in `MyrialArmorItem`? Should remove it from there.

**Review by MattKirchh — DISMISSED** [review]

File: src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java (hunk 1 above)
Comment (insertStringHere) [reply src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java]: Fixed; the code actually needed to be significantly expanded to work; should be fixed now

**Comment by MattKirchh** [comment]
I got some feedback from the private release about the ordering of items in the creative menu. Since that's controlled by the registries, do you want to address the feedback here? I've copied the relevant points of feedback below.

  - The order in which the ingots are listed does not line up with the order of the tools and they should not be separated by other items (alien circuitry, energized focus, etc.)
  - The order of tools in the creative menu does not match vanilla, vanilla goes sword -> shovel -> pickaxe -> axe -> hoe
  - Blocks are listed before items in the creative menu in vanilla, MineBound blocks are listed after MineBound items, blocks should be first
  - Armors (energized iron and illuminant steel) are separated from their tool counterparts, should probably be all tools immediately followed by all armors with myrial armor at the end

**Comment by insertStringHere** [comment]
> I got some feedback from the private release about the ordering of items in the creative menu. Since that's controlled by the registries, do you want to address that feedback here? I've copied the feedback below.
> 
> * The order in which the ingots are listed does not line up with the order of the tools and they should not be separated by other items (alien circuitry, energized focus, etc.)
> * The order of tools in the creative menu does not match vanilla, vanilla goes sword -> shovel -> pickaxe -> axe -> hoe
> * Blocks are listed before items in the creative menu in vanilla, MineBound blocks are listed after MineBound items, blocks should be first
> * Armors (energized iron and illuminant steel) are separated from their tool counterparts, should probably be all tools immediately followed by all armors with myrial armor at the end

If you look, the order in the registries is correct; I'm not sure why it's not working. I'll do some research

**Comment by MattKirchh** [comment]
> > I got some feedback from the private release about the ordering of items in the creative menu. Since that's controlled by the registries, do you want to address that feedback here? I've copied the feedback below.
> > 
> > * The order in which the ingots are listed does not line up with the order of the tools and they should not be separated by other items (alien circuitry, energized focus, etc.)
> > * The order of tools in the creative menu does not match vanilla, vanilla goes sword -> shovel -> pickaxe -> axe -> hoe
> > * Blocks are listed before items in the creative menu in vanilla, MineBound blocks are listed after MineBound items, blocks should be first
> > * Armors (energized iron and illuminant steel) are separated from their tool counterparts, should probably be all tools immediately followed by all armors with myrial armor at the end
> 
> If you look, the order in the registries is correct; I'm not sure why it's not working. I'll do some research

The order in the registries does match what is shown in-game, but it does not follow vanilla convention.

File: src/main/java/com/mineboundteam/minebound/MineBound.java [hunk 2]
```diff
@@ -28,6 +27,13 @@ public class MineBound {
             return new ItemStack(ItemRegistry.MYRI_MANASAC.get());
         }
     };
+    public static final CreativeModeTab SPELLS_TAB = new CreativeModeTab(MOD_ID) {
+        @OnlyIn(Dist.CLIENT)
+        @Override
+        public @NotNull ItemStack makeIcon() {
+            return new ItemStack(ItemRegistry.TELEKINETIC_OFFENSIVE_1.get());
+        }
+    };
```
Comment (MattKirchh) [inline src/main/java/com/mineboundteam/minebound/MineBound.java:36]: Is there some way we can rename the tabs? They're currently both named `MineBound`.

Comment (insertStringHere) [reply src/main/java/com/mineboundteam/minebound/MineBound.java:36]: Yeah, I will; I'm going to hardcode the order, more or less

**Comment by insertStringHere** [comment]
> > > I got some feedback from the private release about the ordering of items in the creative menu. Since that's controlled by the registries, do you want to address that feedback here? I've copied the feedback below.
> > > 
> > > * The order in which the ingots are listed does not line up with the order of the tools and they should not be separated by other items (alien circuitry, energized focus, etc.)
> > > * The order of tools in the creative menu does not match vanilla, vanilla goes sword -> shovel -> pickaxe -> axe -> hoe
> > > * Blocks are listed before items in the creative menu in vanilla, MineBound blocks are listed after MineBound items, blocks should be first
> > > * Armors (energized iron and illuminant steel) are separated from their tool counterparts, should probably be all tools immediately followed by all armors with myrial armor at the end
> > 
> > 
> > If you look, the order in the registries is correct; I'm not sure why it's not working. I'll do some research
> 
> The order in the registries does match what is shown in-game, but it does not follow vanilla convention.

- Ingot order has been fixed
- Tool order has been fixed
- Blocks have been hardcoded to sort to front of main tab
- This is how it is listed in the Combat tab of vanilla. Order has been set to tools, armor, myrial armor

**Review by MattKirchh — APPROVED** [review]

**Pull request merged**